src/main/java/ch/njol/skript/classes/data/JavaClasses.java

```diff
@@ -31,15 +13,20 @@
 import ch.njol.skript.util.Utils;
 import ch.njol.util.StringUtils;
 import ch.njol.yggdrasil.Fields;
+import org.jetbrains.annotations.Contract;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.function.Function;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class JavaClasses {
 
 	public static final int VARIABLENAME_NUMBERACCURACY = 8;
-	public static final Pattern INTEGER_PATTERN = Pattern.compile("-?[0-9]+");
-	public static final Pattern NUMBER_PATTERN = Pattern.compile("-?[0-9]+(?>\\.[0-9]+)?%?");
+	public static final Pattern INTEGER_PATTERN =
+		Pattern.compile("(?<num>-?[0-9]+)( (in )?((?<rad>rad(ian)?s?)|deg(ree)?s?))?");
```

Comment:
```suggestion
		Pattern.compile("(?<num>-?[0-9]+)(?: (?:in )?(?:(?<rad>rad(?:ian)?s?)|deg(?:ree)?s?))?");
```
use non-capturing where possible

Reply:
with the complexity given, would also be a good time to add documentation explaining the regex.